community/ndp/messaging-v1/src/main/java/org/neo4j/ndp/messaging/v1/PackStreamMessageFormatV1.java

```diff
@@ -59,6 +61,45 @@
     public static final byte RELATIONSHIP = 'R';
     public static final byte PATH = 'P';
 
+    // The forward and backward step structures represent
+    // directed relationships within a path but do not
+    // contain identities for the start and end nodes of
+    // those relationships. This removes duplication of
+    // information within path structures, thereby reducing
+    // the overall size of a path's serialised representation.
+    //
+    // Example path:
+    //
+    //     (alice)-[:KNOWS]->(bob)<-[:KNOWS]-(carol)
+    //
+    // Structured representation:
+    //
+    // Struct 'P' Path {
+    //     Struct 'N' Node {           // (alice)
+    //         Identity, Labels, Properties
+    //     }
+    //     Struct '>' ForwardStep {    // -[:KNOWS]->
```

Comment:
I like this approach a lot - but are there perhaps different vocab we could use here? 'ForwardStep'/'BackwardStep' sounds like they pertain to the direction the traversal happened in (stepping forwards and backwards along the path sounds like retracing earlier steps).

The vocabulary we use elsewhere for talking about the direction of relationships from the pov of one of the nodes is `incoming` and `outgoing`, maybe we can use that here? 

Something like `IncomingRelationship` and `OutgoingRelationship` or something similar is explicit, and highlights that these are relationships represented relative to something.